lib/resources/postgres.rb

```diff
@@ -39,16 +39,17 @@ def determine_dirs
         # installed as well as multiple "clusters" to be configured.
         #
         @version = version_from_psql || version_from_dir('/etc/postgresql')
-        @cluster = cluster_from_dir("/etc/postgresql/#{@version}")
-        @conf_dir = "/etc/postgresql/#{@version}/#{@cluster}"
-        @data_dir = "/var/lib/postgresql/#{@version}/#{@cluster}"
+        if !@version.to_s.empty?
+          @cluster = cluster_from_dir("/etc/postgresql/#{@version}")
+          @conf_dir = "/etc/postgresql/#{@version}/#{@cluster}"
+          @data_dir = "/var/lib/postgresql/#{@version}/#{@cluster}"
+        end
       else
         @version = version_from_psql
-        if @version.nil?
+        if @version.to_s.empty?
```

Comment:
Is the concern that it _could_ be an empty string instead of nil?

Reply:
Yes, current and future changes of version parsing could provide a nil or empty string. Wanted to guard against both while at it.

Reply:
Having the version *parsing* come back nil is not a sign of an _error_?